packages/logger/src/services/json-encoder.spec.ts

```diff
@@ -0,0 +1,399 @@
+      })
+
+      it('encodes object with an empty array value without newlines', () => {
+        const encoder = createUglyJsonEncoder()
+        const myObject = { list: [] }
+
+        expect(encoder.encode(myObject)).toBe('{"list":[]}')
+      })
+
+      it('encodes object with an number[] value without newlines', () => {
+        const encoder = createUglyJsonEncoder()
+        const myObject = { list: [1,2,3,4] }
+
+        expect(encoder.encode(myObject)).toBe('{"list":[1,2,3,4]}')
+      })
+
+      it('encodes object with empty object and empty array values without newlines', () => {
+        const encoder = createUglyJsonEncoder()
+        const myObject = { emptyObject: {}, emptyList: [] }
+
+        expect(encoder.encode(myObject)).toBe('{"emptyObject":{},"emptyList":[]}')
+      })
+
+      it('encodes nested objects without newlines', () => {
+        const encoder = createUglyJsonEncoder()
+        const myObject = { nestedObject: { hello: 'world' } }
+
+        expect(encoder.encode(myObject)).toBe('{"nestedObject":{"hello":"world"}}')
+      })
+
+      it('encodes empty array as []', () => {
+        const encoder = createUglyJsonEncoder()
+        const emptyList: string[] = []
+
+        expect(encoder.encode(emptyList)).toBe('[]')
+      })
+
+      it('encodes number array without newlines', () => {
+        const encoder = createUglyJsonEncoder()
+        const myList: number[] = [1, 2, 3, 4]
+
+        expect(encoder.encode(myList)).toBe('[1,2,3,4]')
+      })
+
+      it('encodes string array without newlines', () => {
+        const encoder = createUglyJsonEncoder()
+        const myList: string[] = ['hello', 'there', 'world']
+
+        expect(encoder.encode(myList)).toBe('["hello","there","world"]')
+      })
+
+      it('encodes bigint array as number array without newlines', () => {
+        const encoder = createUglyJsonEncoder()
+        const myList: bigint[] = [1n, 2n, 3n, 4n]
+
+        expect(encoder.encode(myList)).toBe('[1,2,3,4]')
+      })
+
+      it('encodes error array as string array without newlines', () => {
+        const encoder = createUglyJsonEncoder()
+        const myList: Error[] = [new Error('my error')]
+
+        expect(encoder.encode(myList)).toMatch(/^\["Error: my error\\n +at.+"\]$/)
+      })
+
+      it('encodes empty object value in array without newlines', () => {
+        const encoder = createUglyJsonEncoder()
+        const myList: object[] = [{}]
+
+        expect(encoder.encode(myList)).toBe('[{}]')
+      })
+
+      it('encodes object value in array without newlines', () => {
+        const encoder = createUglyJsonEncoder()
+        const myList: object[] = [{ hello: 'world' }]
+
+        expect(encoder.encode(myList)).toBe('[{"hello":"world"}]')
+      })
+
+    })
+
+    describe('pretty', () => {
+
+      it('encodes undefined', () => {
+        const encoder = createUglyJsonEncoder()
```

Comment:
You are in a 'pretty' describe, so why is the `createUglyJsonEncoder()` used?
There is a couble of these, down below.

Reply:
If it is because they are encoded the same between ugly/pretty, should using the `createPrettyJsonEncoder()` not yield the same result?

If it is because they are encoded differently between ugly/pretty, then are you sure that this is not a specific case, where the styling should be unified between ugly/pretty, as the input contains little/no information?

Reply:
When writing the tests, I started with the ugly ones. Then I copied them over to the pretty describe, changed them. So these were just typoes. Good catch :rocket: